src/bench/bench.cpp

```diff
@@ -103,30 +130,24 @@ void BenchRunner::RunAll(const Args& args)
             continue;
         }
 
-        Bench bench;
-        if (args.sanity_check) {
-            bench.epochs(1).epochIterations(1);
-            bench.output(nullptr);
-        }
-        bench.name(name);
-        if (args.min_time > 0ms) {
-            // convert to nanos before dividing to reduce rounding errors
-            std::chrono::nanoseconds min_time_ns = args.min_time;
-            bench.minEpochTime(min_time_ns / bench.epochs());
-        }
-
-        if (args.asymptote.empty()) {
-            func(bench);
-        } else {
-            for (auto n : args.asymptote) {
-                bench.complexityN(n);
-                func(bench);
+        if (properties.iterate_sha256_implementations) {
```

Comment:
Hm, I don't really like that all of that SHA256 specific code is becoming part of the benchmark framework. Wouldn't it be sufficient to just adapt the SHA benchmark in `crypto_hash.cpp` so that there is a separate test for each of the implementations? There could be a separate benchmark for each value of the `enum UseImplementation`.

Reply:
> Hm, I don't really like that all of that SHA256 specific code is becoming part of the benchmark framework. 

Yea. Concept NACK on the current approach.